docs/source/format/Intro.rst

```diff
@@ -0,0 +1,510 @@
+
+   Physical layout diagram for dictionary data type.
+
+Run-End Encoded Layout
+======================
+
+Run-end encoding is well-suited for representing data containing sequences of the
+same value. These sequences are called runs. Run-end encoded array has no buffers
+by itself, but has two child arrays:
+
+*  **Run ends array:** holds the index in the array where each run ends.
+*  **Values array:** the actual values without repetitions.
+
+.. figure:: ./images/ree-diagram.svg
+   :alt: Diagram is showing the difference between the run-end encoded data
+         type presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for run-end encoded data type.
+
+.. link to All types overview https://github.com/apache/arrow/issues/14752
+
+Extension Types
+===============
+
+In case the system or application needs to extend standard Arrow data types with
+custom semantics this is enabled by defining extension types or user-defined types.
+
+For example:
+
+* Universally unique identifiers (UUID) can be represented as a FixedSizeBinary type
+* Trading time can be represented as a Timestamp with metadata indicating the market
+  trading calendar
+
+Extension types can be defined by annotating any of the built-in Arrow data types
+(the “storage type”) with a custom type name and optional serialized representation
+(``'ARROW:extension:name'`` and ``'ARROW:extension:metadata'`` keys in the Field
+metadata structure).
+
+.. seealso::
+   The :ref:`format_metadata_extension_types` documentation.
+
+Canonical Extension Types
+-------------------------
+
+It is beneficial to share the definitions of well-known extension types so as to
+improve interoperability between different systems integrating Arrow columnar data.
+For this reason canonical extension types are defined in Arrow itself.
+
+Examples:
+
+* Fixed and variable shape tensor
+
+  - :ref:`fixed_shape_tensor_extension`
+  - :ref:`variable_shape_tensor_extension`
+
+.. seealso::
+   The :ref:`format_canonical_extensions` documentation.
+
+Community Extension Types
+-------------------------
+These are Arrow extension types that have been established as standards within specific domain areas.
+
+Example:
+
+* `GeoArrow`_: A collection of Arrow extension types for representing vector geometries
+
+.. _GeoArrow: https://github.com/geoarrow/geoarrow
+
+The Arrow C Data Interface
+==========================
+
+Arrow memory layout is meant to be a universal standard for tabular data, not tied to a specific
+implementation.
+
+While there are specifications to share Arrow data between processes or over the network (e.g. the
+IPC messages), the Arrow C Data Interface is meant to actually zero-copy share the data between
+different libraries within the same process (i.e. actually share the same buffers in memory).
+
+The Arrow C Data Interface defines a set of small C structures:
+
+.. code-block::
+
+   struct ArrowSchema {
+    const char* format;
```

Comment:
For the tutorial during the conference, I included the struct to explain things. But for the intro here I am not sure we should include it without much more explanation of the details of those structs.

For the purpose of this intro page, I would maybe also not call out the C Data Interface specifically, but we could maybe list and briefly explain the additional specifications on top of the columnar format, which could be (for this intro) the C Data Interface to share data zero-copy within the same process and IPC as serialization format for between processes